doc/embedded/crosscompile_example.md

```diff
@@ -0,0 +1,132 @@
+## Crosscompile mlpack example for an embedded hardware
+
+In this article, we explore how to add mlpack to a CMake project that cross-compiles
+code to embedded hardware.  See also these related other guides, which may be useful
+to read before this one:
+ * [Run mlpack bindings on a Raspberry Pi](crosscompile_armv7.md)
+ * [Set up cross-compilation toolchain for mlpack](supported_boards.md)
+
+
+### Cloning mlpack example respository
+
+mlpack has an [example repository](https://github.com/mlpack/examples) that
+
+shows a number of applications and use cases for mlpack, including embedded
+deployment.  In this tutorial, we are interested in the `embedded/` directory,
+which provides a CMake project template that compiles a random forest
+application to embedded hardware.  This project template can be adapted to a new
+project, or its pieces can be incorporated into an existing CMake project.
+
+The first step is to clone the examples repository:
+
+```sh
+git clone git@github.com:mlpack/examples.git
+```
+
+Next, let's look at the `CMakeLists.txt` (e.g. the CMake configuration) in the
+`embedded/crosscompile_random_forest/` directory.
+
+### `CMakeLists.txt`
+
+The first part of the code, printed below or
+[available here](https://github.com/mlpack/examples/blob/master/embedded/crosscompile_random_forest/CMakeLists.txt),
+defines the project name and includes two useful CMake configuration files:
+ * `mlpack.cmake`: finds mlpack's dependencies and download them if necessary.
+ * `ConfigureCrossCompile.cmake`: set up CMake configuration for cross-compilation.
+ * `crosscompile-toolchain.cmake`: invoke CMake crosscompilation infrastructure.
+ * `crosscompile-arch-config.cmake`: add necessary flags depending on the
+   architecture (optional).
+
+Then we need to call `fetch_mlpack()` will download mlpack including all dependencies,
+cross-compile openblas and set up all the necessary parameters to find these dependencies.
+Most of mlpack dependencies are header-only with the exception of Openblas,
+thus this is expected to be a quick step.
+
+`fetch_mlpack()` will detect if cross compilation is necessary or not depending
+on the command that is executed when running cmake. Based on this, it will
+compile OpenBLAS.
+
+```cmake
+cmake_minimum_required(VERSION 3.11)
+project(main)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake")
+include(CMake/Autodownload.cmake)
+include(CMake/ConfigureCrossCompile.cmake)
```

Comment:
I think these can be updated; at least, Autodownload.cmake isn't needed anymore.